File: backend/src/api/services/Item.ts
class Item {
    name: string;
    description: string;
    img: string;
    groceryStoreName: string;
    distance: number;
    price: number;
    savings: number;

    constructor(name: string, description: string, img: string, groceryStoreName: string, distance: number, price: number) {
        this.name = name;
        this.description = description;
        this.img = img;
        this.groceryStoreName = groceryStoreName;
        this.distance = distance;
        this.price = price;
        this.savings = 0;
    }
}

export { Item };

File: frontend/budget_basket/lib/models/item.dart
import 'package:budget_basket/models/grocery_store.dart';

class Item {
  final String itemName;
  final String? itemDescription;
  final GroceryStore itemStore;
  final double itemDistance;
  final double itemPrice;
  final double? itemSavings;
  final String? itemImgUrl;

  Item({
    required this.itemName,
    this.itemDescription,
    required this.itemStore,
    required this.itemDistance,
    required this.itemPrice,
    this.itemSavings,
    this.itemImgUrl,
  });

  factory Item.fromJson(Map<String, dynamic> json) {
    print("ITEM: $json");
    return Item(
      itemName: json['name'],
      itemDescription: json['description'],
      itemStore: GroceryStore.values.firstWhere(
          (e) => e.toString() == 'GroceryStore.${json['groceryStoreName']}'),
      itemDistance: json['distance'],
      itemPrice: json['price'],
      itemSavings: null,
      itemImgUrl: json['img'],
    );
  }
}
